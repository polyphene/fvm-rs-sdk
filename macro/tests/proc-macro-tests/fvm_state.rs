#![allow(unreachable_code)]
use fvm_rs_sdk::*;

#[fvm_state]
pub struct MockStruct1 {
    pub count: u64,
}

#[fvm_state]
pub struct MockStruct2<'a> {
    pub count: &'a u64,
}

fn main() {}
