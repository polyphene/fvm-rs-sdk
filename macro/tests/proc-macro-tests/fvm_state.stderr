error: structs with #[fvm_state] cannot have lifetime or type parameters
  --> tests/proc-macro-tests/fvm_state.rs:10:23
   |
10 | pub struct MockStruct2<'a> {
   |                       ^^^^
